backbone.js

```diff
@@ -1455,6 +1455,16 @@
 
   // Set up all inheritable **Backbone.Router** properties and methods.
   _.extend(Router.prototype, Events, {
+    // If set to true, route handlers will be called with routeData object
+    //
+    //     this.route('search/:query/p:num', 'search', function(routeData) {
```

Comment:
@akre54 
> I don't know about the routeData object though. How about (params, queryString, name) as the arguments to the handler? Or (params, routeData) where routeData is queryString and name.

I think like `(params, routeData)` more. Having `routeData` as object gives option to expand the object with more stuff to help the route handler if needed, few things I could think of:
- `query` - object representation of queryString: 'x=true&y=5&z=abc' => `{x: "true", y: "5", z: "abc"}`.
  - Another option to enable queryString parsing maybe?
  - @jmeas is already doing this in [Backbone Router](https://github.com/jmeas/backbone.base-router#query) 
- `fragment`